### DISCUSSION TITLE
`examples/compose/local/setup.sh` configures an incorrectly formatted password hash

### DISCUSSION THREAD

**Discussion by MaxTrautwein** [authelia/authelia#4642] [Q&A] [answered]
### Version

v4.37.5

### Deployment Method

Docker

### Reverse Proxy

Traefik

### Reverse Proxy Version

2.9.6

### Description

The automatic example Setup via https://github.com/authelia/authelia/blob/master/examples/compose/local/setup.sh Fails.
Navigating to the  sites Displays a 404 Error.

The cause is that the `setup.sh` places an incorrectly formatted password into the `authelia/users_database.yml` File.
Running the in the script used command for generationg the Password yields:
```
$ sudo docker run authelia/authelia authelia hash-password 'secure' | sed 's/Password hash: //g'
Digest: $argon2id$v=19$m=65536,t=3,p=4$tIyFT7ArH4yXqVjznDbc1Q$1JuIiK74aZD1O5ZD+c/F5JsE50HANDMK5W4z+QIl0Ts
```

`Digest: $argon2id$v=19$m=65536,t=3,p=4$tIyFT7ArH4yXqVjznDbc1Q$1JuIiK74aZD1O5ZD+c/F5JsE50HANDMK5W4z+QIl0Ts`
is then placed as the hash inside of `authelia/users_database.yml`
where `4$tIyFT7ArH4yXqVjznDbc1Q$1JuIiK74aZD1O5ZD+c/F5JsE50HANDMK5W4z+QIl0Ts` 
Would have been expected / Works.

### Reproduction

1. Clone the Repository
2. Checkout version 4.37.5
3. navigate to `examples/compose/local`
4. Execute `setup.sh`
5. Enter the Data
6. Follow any of the displayed links

### Logs (Authelia)

```shell
time="2022-12-26T00:37:31+11:00" level=fatal msg="The following providers had fatal failures during startup: user" stack="github.com/authelia/authelia/v4/internal/commands/root.go:309 doStartupChecks\ngithub.com/authelia/authelia/v4/internal/commands/root.go:87  cmdRootRun\ngithub.com/spf13/cobra@v1.6.1/command.go:920                  (*Command).execute\ngithub.com/spf13/cobra@v1.6.1/command.go:1044                 (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.6.1/command.go:968                  (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10       main\nruntime/proc.go:250                                           main\nruntime/asm_amd64.s:1594                                      goexit"
time="2022-12-26T00:37:14+11:00" level=error msg="Failure running the user provider startup check: error decoding the authentication database: failed to parse hash for user 'max': provided encoded hash has an invalid format: the digest doesn't begin with the delimiter '$' and is not one of the other understood formats" stack="github.com/authelia/authelia/v4/internal/commands/root.go:287 doStartupChecks\ngithub.com/authelia/authelia/v4/internal/commands/root.go:87  cmdRootRun\ngithub.com/spf13/cobra@v1.6.1/command.go:920                  (*Command).execute\ngithub.com/spf13/cobra@v1.6.1/command.go:1044                 (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.6.1/command.go:968                  (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10       main\nruntime/proc.go:250                                           main\nruntime/asm_amd64.s:1594                                      goexit"
```

### Pre-Submission Checklist

- [X] I agree to follow the [Code of Conduct](http://www.authelia.com/code-of-conduct)
- [X] This is a bug report and not a support request
- [X] I have read the security policy and this bug report is not a security issue or security related issue
- [X] I have either included the complete configuration file or I am sure it's unrelated to the configuration
- [X] I have provided all of the required information in full with the only alteration being reasonable sanitization in accordance with the [Troubleshooting Sanitization](https://www.authelia.com/r/sanitize) reference guide
- [X] I have checked for related proxy or application logs and included them if available
- [X] I have checked for related issues and checked the documentation

**Comment by james-d-elliott** [comment]
The issues board is for issues with the application, not issues with the docs. Will fix this up shortly.

**Reply by avicoder** [reply]
This issue still persist.

**Reply by james-d-elliott** [reply]
I don't believe that's accurate: https://github.com/authelia/authelia/commit/3ecb9f1d44643a5f9f4c4e78c8dcbb0d1d29b074

**Reply by ReineL** [reply]
The setup.sh has been fixed, but the problem persists if you follow the documentation on deployment with docker:
```
git clone https://github.com/authelia/authelia.git
cd authelia
git checkout $(git describe --tags `git rev-list --tags --max-count=1`)
```
The checkout overwrites authelia/examples/compose/local/setup.sh with an older version.

**Comment by james-d-elliott** [comment, marked as answer]
Can  temporarily be worked around by  doing:

```
git clone https://github.com/authelia/authelia.git
cd authelia
git checkout 3ecb9f1
```
